lib/game.rb

```diff
@@ -0,0 +1,94 @@
+class Game
+  attr_accessor :board, :player_1, :player_2  #provides access to the board, player_1, player_2
+
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new ) #Game initialize defaults to two human players, X and O, with an empty board
+    @board = board                #Game initialize accepts 2 players and a board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  WIN_COMBINATIONS = [      #defines a constant WIN_COMBINATIONS with arrays for each win combination
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [6,4,2],
+    [0,4,8]
+  ]
+
+  def over? #returns true for a draw or a won game
+    won? || draw?
+  end
+
+  def won?  #returns false for a draw
+    WIN_COMBINATIONS.detect do |combo|
+      first_index = combo[0]
+      second_index = combo[1]
+      third_index = combo[2]
+      cell_one = board.cells[first_index]
+      cell_two = board.cells[second_index]
+      cell_three = board.cells[third_index]
+
+      if cell_one == "X" && cell_two == "X" && cell_three == "X" || cell_one == "O" && cell_two == "O" && cell_three == "O"
+        true  #returns the correct winning combination in the case of a win
+      else
+        false #returns false for a draw
+      end
+    end
+  end
+
+  def draw?
+    if board.full? && !won?
+      true  #returns true for a draw, returns false for a won game or an in-progress game
+    end
+  end
+
+  def winner
+    #returns X when X won
+    !won? ? nil : board.cells[won?.first]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.